lib/api/api.dart

```diff
@@ -779,4 +793,10 @@ class ApiSingleton {
       return false;
     }
   }
+  
+  Future<String> getCityNameFromCoords(double lat, double lon) async {
+    var locale = '${await UserManager.getLanguage()}_${await UserManager.getLanguageCountry()}';
```

Comment:
Locale in geocoding is optional, if not provided it uses the one that the phone has selected
If we add it, it has to be in the formal language_country (ie es_ES, de_DE, en_US...)

Reply:
Would you mind creating a function in the UserManger called getUserLocale (or something like this) that returns this value? It could be worth reusing that function in the future for other cases instead of contacting the result of getLanguage and getLanguageCountry always

Reply:
It would be nice to check if it works in basque... I was wondering if eu_ES is the correct locale code. If it's prone to error, only using language is also accepted in the function `placemarkFromCoordinates`

Reply:
> Would you mind creating a function in the UserManger called getUserLocale (or something like this) that returns this value?

done

> It would be nice to check if it works in basque... I was wondering if eu_ES is the correct locale code. If it's prone to error, only using language is also accepted in the function placemarkFromCoordinates

Seems like it's not working as expected. It's returning the name in the countries name instead of in selected locale (Mountain View in US; München in Germany, and Bilbo in Basque), when I have basque as my language in the app
![image](https://github.com/Mosquito-Alert/Mosquito-Alert-Mobile-App/assets/157690872/c28d0559-9f9e-41b7-92aa-c03f4a263f08)

Reply:
> Or try using official method: https://api.flutter.dev/flutter/intl/Intl/getCurrentLocale.html

```
static String getCurrentLocale() {
  return defaultLocale ??= systemLocale;
}
```

But then with this method, if the user selects another language within the app, it will always still show the systemLocale

Reply:
Ah yes, you're right! The only thing I'm afraid of is the `UserManger.getUserLocale` returning a combination of language + country that is not a valid locale and the code fails. Maybe it's safer to just use the language code as stated in the [docu](https://pub.dev/packages/geocoding) ('en' for English)

Reply:
> The localeIdentifier should be formatted using the syntax: [languageCode]_[countryCode]

`en` seems to be the only option where a country code is not needed. If we are afraid, maybe the safest option is to leave this option parameter empty, so it will use the system's language

Reply:
As far as this codes are always valid, it's gonna be ok
https://github.com/Mosquito-Alert/Mosquito-Alert-Mobile-App/blob/6c5993a230a86f958c8dca8bcfef2994a6b93ebe/lib/pages/settings_pages/settings_page.dart#L38-L62